Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/HttpObservable.java

```diff
@@ -347,7 +360,10 @@ private SSLConnectionSocketFactory createSSLSocketFactory() {
             URIBuilder builder = new URIBuilder(httpRequest.getURI());
 
             // Add parameters
-            builder.setParameters(getDefaultParameters());
+            // check parameters size to avoid add ? at uri end which can cause invalid uri
+            if(getDefaultParameters().size() > 0){
```

Comment:
>if(getDefaultParameters().size() > 0){ [](start = 12, length = 38)

No need

Reply:
it will auto add ? at the end of uri which causes invalid urrl for getting accesskey 

---
In reply to: [271137782](https://github.com/Microsoft/azure-tools-for-java/pull/2961#discussion_r271137782) [](ancestors = 271137782)